### PR TITLE
Feat/extras

### PR DESCRIPTION
**Pull request by Aditi0505**
# Change Logs:

## Description:

- Added Theme
- Added custom responsiveness
- Added manifest file information 
- Added readme
## Screenshot 

image
image

# Preview Link
https://deploy-preview-4--afterglow.netlify.app/

### CODE REVIEW

**Review by ranitasahaa1230 — APPROVED** [review]
Clean UI
Code can be understood easily
Loved the todo component
Looks good to merge

**Comment by Aditi0505** [comment]
> Clean UI Code can be understood easily Loved the todo component Do add `_redirects` file Rest, looks good

Thank you!

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
 REACT_APP_API_KEY=b232e0899243d155415284ed1c81219b
+INLINE_RUNTIME_CHUNK=false
```
Comment (rohanmathur91) [inline .env:2]: you should not push `.env` on github

File: src/App.js [hunk 2]
```diff
@@ -7,30 +7,35 @@ import {
...
-      </TodoProvider>
-    </FilterProvider>
+    <div className={themeState.theme === "dark" ? "dark" : ""}>
+      <FilterProvider>
+        <TodoProvider>
+          <TimeProvider>
+            <WeatherProvider>
+              <QuoteProvider>
+                <UserProvider>
+                  <WallpaperProvider>
```
Comment (rohanmathur91) [inline src/App.js:25]: suggestion: you can reduce the number of contexts, try to figure out that which functionalities can be together, and have one context for them.

File: src/pages/User-OnBoard/WelcomeUser.jsx [hunk 3]
```diff
@@ -36,7 +37,14 @@ const WelcomeUser = () => {
...
           {!isEdit ? (
             <h2 className="w-full flex justify-center items-center group font-Montserrat mt-5">
-              Good Evening, {userName}.
+              {currentTime.split(":")[0] >= "00" &&
+              currentTime.split(":")[0] <= "12"
+                ? `Good Morning, ${userName}.`
+                : currentTime.split(":")[0] >= "13" &&
+                  currentTime.split(":")[0] <= "15"
+                ? ` Good Afternoon, ${userName}.`
+                : `Good Evening, ${userName}.`}
```
Comment (rohanmathur91) [inline src/pages/User-OnBoard/WelcomeUser.jsx:46]: have a function which does this and returns the required string as you are using it in multiple places

File: src/reducer/themeReducer.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const themeReducer = (state, action) => {
+    switch (action.type) {
+      case "DARK":
+        return { ...state, theme: "light" };
+      case "LIGHT":
+        return { ...state, theme: "dark" };
+      default:
+        return state;
+    }
+  };
```
Comment (rohanmathur91) [inline src/reducer/themeReducer.js:10]: no need of a reducer to manage the theme state, you can do this with a single `useState`.

**Review by rohanmathur91 — APPROVED** [review]
- Good use of tailwind, added a few suggestions have a look.
- Great work.

File: src/App.js (hunk 2 above)
Comment (Aditi0505) [reply src/App.js:25]: Sure, I will try to figure this out later.

File: .env (hunk 1 above)
Comment (Aditi0505) [reply .env:2]: Sure, I will keep it in mind now onwards.

File: src/pages/User-OnBoard/WelcomeUser.jsx (hunk 3 above)
Comment (Aditi0505) [reply src/pages/User-OnBoard/WelcomeUser.jsx]: Thank you, I have made the necessary changes.

File: src/reducer/themeReducer.js (hunk 4 above)
Comment (Aditi0505) [reply src/reducer/themeReducer.js]: Thank you, I have made the necessary changes.

**Pull request merged**